plugins/dynamix.docker.manager/include/DockerClient.php

```diff
@@ -336,15 +336,16 @@ public function getAllInfo($reload=false,$com=true,$communityApplications=false)
 		return $info;
 	}
 
-	public function getIcon($Repository,$contName,$tmpIconUrl='') {
+	public function getIcon($Repository,$contName,$iconUrl='') {
 		global $docroot, $dockerManPaths;
-		$imgUrl = $this->getTemplateValue($Repository, 'Icon','all',$contName);
-		if (!$imgUrl) $imgUrl = $tmpIconUrl;
+		$imgUrl = $iconUrl ?: $this->getTemplateValue($Repository, 'Icon','all',$contName);
 		if (!$imgUrl || trim($imgUrl) == "/plugins/dynamix.docker.manager/images/question.png") return '';
 
-		$imageName = $contName ?: $name;
-		$iconRAM = sprintf('%s/%s-%s.png', $dockerManPaths['images-ram'], $contName, 'icon');
-		$icon    = sprintf('%s/%s-%s.png', $dockerManPaths['images'], $contName, 'icon');
+		$imgUrlHash = sha1($imgUrl);
+		$iconFile = sprintf('%s-%s.png', 'icon', $imgUrlHash);
```

Comment:
Instead of storing the icons by container name only this change uses a sha1 hash of the imgUrl. This way a change to the icon url will automatically invalidate the cached icon.